spark-extension-shims-spark3/src/main/scala/org/apache/spark/sql/blaze/ShimsImpl.scala

```diff
@@ -562,7 +562,20 @@ class ShimsImpl extends Shims with Logging {
         val inputRDD = executeNative(child)
         val nativeShuffle = getUnderlyingNativePlan(child).asInstanceOf[NativeShuffleExchangeExec]
         val nativeSchema: pb.Schema = nativeShuffle.nativeSchema
-        val metrics = MetricNode(Map(), inputRDD.metrics :: Nil)
+
+        val requiredMetrics = nativeShuffle.readMetrics ++
+          nativeShuffle.metrics.filterKeys(_ == "shuffle_read_total_time")
+        val metrics = MetricNode(
+          requiredMetrics,
+          inputRDD.metrics :: Nil,
+          Some({
+            case ("output_rows", v) =>
```

Comment:
output_rows is counted in shulffle writer, is it necessary to count it again in shuffle reader?

Reply:
do you mention here?
https://github.com/kwai/blaze/blob/e7270edf597c2f340df0937a92358917eda51045/spark-extension/src/main/scala/org/apache/spark/sql/execution/blaze/plan/NativeShuffleExchangeBase.scala#L195-L197
From my observation, they are not the same. Suppose there is a stage like this `shuffle_read -> final_agg -> shuffle_write`.
`output_row` counted in `ShimsImpl.executeNativeAQEShuffleReader` is for `shuffle_read` or `IpcReaderExec`
`output_row` counted in `NativeShuffleExchangeBase.prepareNativeShuffleDependency` is for `shuffle_write` or `ShuffleWriterExec`
If we do not count `output_row` in `ShimsImpl.executeNativeAQEShuffleReader`, the shuffle read records metrics is missing as I mention in  #872.